include/z64environment.h

```diff
@@ -132,17 +132,36 @@ typedef struct {
     /* 0x06 */ u8 skybox2Index;
 } TimeBasedSkyboxEntry; // size = 0x8
 
+// Get blend rate from `EnvLightSettings.blendRateAndFogNear` in 0-255 range
+#define ENV_LIGHT_SETTINGS_BLEND_RATE_U8(blendRateAndFogNear) (((blendRateAndFogNear) >> 10) * 4)
+#define ENV_LIGHT_SETTINGS_FOG_NEAR(blendRateAndFogNear) ((blendRateAndFogNear) & 0x3FF)
+
 typedef struct {
     /* 0x00 */ u8 ambientColor[3];
     /* 0x03 */ s8 light1Dir[3];
     /* 0x06 */ u8 light1Color[3];
     /* 0x09 */ s8 light2Dir[3];
     /* 0x0C */ u8 light2Color[3];
     /* 0x0F */ u8 fogColor[3];
-    /* 0x12 */ s16 fogNear;
-    /* 0x14 */ s16 fogFar;
+    /* 0x12 */ s16 blendRateAndFogNear; // The blendRate determines how fast the current light settings fade to the new one (under LIGHT_MODE_SETTINGS, otherwise unused). The fogNear is in 0-1000 range (0: starts immediately, 1000: no fog), but is clamped to ENV_FOGNEAR_MAX. The fogFar is almost always 1000 (ends at max depth, set by zFar).
```

Comment:
This comment goes way over our line width limit (ie. 120 characters), so I think it should be split in multiple lines, or shortened to fit but I doubt that's an option here

Reply:
I can put the part on fogFar somewhere else but not sure where. Maybe on zFar with some rewording

It would still be too long, and last time this was raised iirc @fig02 didn't want to break comments in structs over several lines?

Reply:
Not sure how it should be handled but I think this is the only part holding the PR so we should probably do it out one way or another

Reply:
I have been meaning to suggest some changes to the comments, but haven't had a moment to write it out.

In short, I think the differences between `EnvLightSettings` and `CurrentEnvLightSettings` are pretty nuanced and need a bit more of a larger explanation, not just a short sentence next to the struct member. imo it warrants a larger comment near the two structs explaining their differences in purpose and the weird quirk about the bitpacked blend rate

When i have a moment to put some thought into it, I can suggest more specific wording

Reply:
sorry this took almost 2 months for me to look at, was just on the backburner for a while lmao

here is my attempt at reorganizing/rewording a bit <https://github.com/fig02/oot/blob/comments/include/z64environment.h#L139>

to try and explain my thought process:
- I moved the "current" struct above the settings struct that holds data that lives in the scene. The "current" lights are more relevant in terms of the light system and what you see in the game. The light settings in the scene can be see as more of a static version that is derived from that

- the "current" struct has explanations for fogNear and zFar (easier imo to have those explanations separate from the confusing `blendRateAndFogNear` variable )

- a longer explanation of blend rate added between the 2 structs, along with explaining why these two very similar structs are separate

Reply:
Looks good to me